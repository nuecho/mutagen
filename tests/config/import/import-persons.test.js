const CONFIGURATION_PATH = getResourcePath("configurations.json");

describe(`[mutagen config import configurations.json]`, () => {
  test(`should import objects when they don't exist`, () => {
    const { code, output } = mutagen(`config import ${CONFIGURATION_PATH}`);

    expect(output).toMatchSnapshot(
      "the two persons have been imported but there are missing dependencies"
    );
    expect(code).toBe(0);
  });

  test(`should do nothing when objects already exist`, () => {
    const { code, output } = mutagen(`config import ${CONFIGURATION_PATH}`);

    expect(output).toMatchSnapshot(
      "the two persons already exist and have not been imported"
    );
    expect(code).toBe(0);
  });
});
