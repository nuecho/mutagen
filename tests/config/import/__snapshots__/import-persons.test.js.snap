// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[mutagen config import configurations.json] should do nothing when objects already exist: the two persons already exist and have not been imported 1`] = `
Object {
  "stderr": "",
  "stdout": "= person => Environment/00001
= person => Environment/00002
Completed. 0 object(s) imported.
",
}
`;

exports[`[mutagen config import configurations.json] should import objects when they don't exist: the two persons have been imported but there are missing dependencies 1`] = `
Object {
  "stderr": "Cannot find script 'Environment/capacityRule'
Cannot find objectivetable 'Environment/contract'
Cannot find place 'Environment/place'
Cannot find folder 'site'
Cannot find skill 'Environment/skill_1'
Cannot find skill 'Environment/skill_2'
Cannot find skill 'Environment/skill_3'
",
  "stdout": "+ person => Environment/00001
+ person => Environment/00002
Completed. 2 object(s) imported.
",
}
`;
